templates/cpp-template-default/CMakeLists.txt

```diff
@@ -134,21 +148,50 @@ set(GAME_HEADERS
   ${PLATFORM_SPECIFIC_HEADERS}
 )
 
+file(GLOB GAME_RESSOURCES "${CMAKE_CURRENT_SOURCE_DIR}/Resources/*.png")
+file(GLOB GAME_RESSOURCES_FONTS "${CMAKE_CURRENT_SOURCE_DIR}/Resources/fonts/*.ttf")
+
+
 if(GAME_HEADERS)
   if ( WIN32 )
     add_executable(${APP_NAME} WIN32 ${GAME_SRC} ${GAME_HEADERS})
+  elseif(MACOSX AND NOT IOS)
+    add_executable(${APP_NAME} MACOSX_BUNDLE ${GAME_SRC} ${GAME_HEADERS} ${GAME_RESSOURCES} ${GAME_RESSOURCES_FONTS})
+    set_source_files_properties(
+      ${GAME_RESSOURCES}
+      PROPERTIES
+      MACOSX_PACKAGE_LOCATION Resources
+      )
+    set_source_files_properties(
+      ${GAME_RESSOURCES_FONTS}
+      PROPERTIES
+      MACOSX_PACKAGE_LOCATION Resources/fonts
+      )
+
   else()
     add_executable(${APP_NAME} ${GAME_SRC} ${GAME_HEADERS})
   endif ( WIN32 )
 else()
   if ( WIN32 )
     add_executable(${APP_NAME} WIN32 ${GAME_SRC})
+  elseif(MACOSX AND NOT IOS)
+    add_executable(${APP_NAME} MACOSX_BUNDLE ${GAME_SRC} ${GAME_RESSOURCES} ${GAME_RESSOURCES_FONTS})
+    set_source_files_properties(
+      ${GAME_RESSOURCES}
+      PROPERTIES
+      MACOSX_PACKAGE_LOCATION Resources
+      )
+    set_source_files_properties(
+      ${GAME_RESSOURCES_FONTS}
+      PROPERTIES
+      MACOSX_PACKAGE_LOCATION Resources/fonts
+      )
   else()
     add_executable(${APP_NAME} ${GAME_SRC})
   endif ( WIN32 )
 endif()
 
-target_link_libraries(${APP_NAME} cocos2d)
+target_link_libraries(${APP_NAME} cocos2d ${PLATFORM_SPECIFIC_LINK_LIBRARIES})
```

Comment:
Is there any case let the variable PLATFORM_SPECIFIC_LINK_LIBRARIES be empty?
If the variable PLATFORM_SPECIFIC_LINK_LIBRARIES is empty, target_link_libraries command could failed, isn't it?